WebAPI/LearningHub.Nhs.Database/Stored Procedures/Hierarchy/HierarchyEditMoveNode.sql

```diff
@@ -279,9 +280,48 @@ BEGIN
 			AND rrdv.Deleted = 0
 			AND hed.Deleted = 0
 
-        -- Remove all instance of the referenced path when moving a referenced folder
+		 -- Remove all instance of the referenced path when moving a referenced folder
 
 		 EXEC hierarchy.HierarchyEditRemoveNodeReferencesOnMoveNode @HierarchyEditDetailId,@UserId,@UserTimezoneOffset
+
+		-- Moving a folder into a referenced folder should affect all instances of the referenced folder.
+
+		DECLARE @CurrentNodeId INT,
+                @ReferenceHierarchyEditDetailId INT
+
+        SELECT @CurrentNodeId = hed.NodeId
+        FROM [hierarchy].[HierarchyEditDetail] hed
+        WHERE hed.Id = @MoveToHierarchyEditDetailId
+		
+        -- Declare the cursor
+
+		 DECLARE NodeCursor CURSOR FOR 
+                  SELECT Id AS ReferenceHierarchyEditDetailId
+                  FROM hierarchy.HierarchyEditDetail
```

Comment:
I got error when running this with my data which are due to the selected HierarchyEditDetail records not being restricted to the current HierarchyEditId.
If there are previous entries (e.g. against discarded edits) with the same NodeId they will be included here.
I think deleted records should also be ignored

Reply:
Done